SonarQube.Common/CommandLine/ArgumentDescriptor.cs

```diff
@@ -14,6 +14,8 @@ namespace SonarQube.Common
     /// </summary>
     public class ArgumentDescriptor
     {
+        public static readonly StringComparer IdComparer = StringComparer.InvariantCultureIgnoreCase;
```

Comment:
Moved from the CommandLineParser class as the descriptor class should specify how its ids should be compare.